Tests/Remove-JiraIssue.Tests.ps1

```diff
@@ -0,0 +1,272 @@
+                  "summary": "Test Sub-Task",
+                  "comment": {
+                    "comments": [],
+                    "maxResults": 0,
+                    "total": 0,
+                    "startAt": 0
+                  }
+                }
+              }
+'@
+
+        }
+
+        Mock Get-JiraConfigServer -ModuleName JiraPS {
+            Write-Output $jiraServer
+        }
+
+        Mock Get-JiraIssue {
+            $obj = $TestIssueJSONs[$Key] | ConvertFrom-Json
+
+            $obj.PSObject.TypeNames.Insert(0, 'JiraPS.Issue')
+
+            $obj | Add-Member -MemberType ScriptMethod -Name ToString -Value {return ""} -Force
+            return $obj
+        }
+
+        Mock Invoke-JiraMethod -ModuleName JiraPS -ParameterFilter {$URI -like "$jiraServer/rest/api/*/issue/TEST-1?*" -and $Method -eq "Delete"} {
+            return $null
+        }
+
+        Mock Invoke-JiraMethod -ModuleName JiraPS -ParameterFilter {$URI -like "$jiraServer/rest/api/*/issue/TEST-2?deleteSubTasks=False" -and $Method -eq "Delete"} {
+            $MockedResponse = @"
+            {
+                "errorMessages": [
+                  "The issue 'TEST-2' has subtasks.  You must specify the 'deleteSubtasks' parameter to delete this issue and all its subtasks."
+                ],
+                "errors": {}
+              }
+"@ | ConvertFrom-Json
+            Resolve-JiraError $MockedResponse -WriteError
+        }
+
+        Mock Invoke-JiraMethod -ModuleName JiraPS -ParameterFilter {$URI -like "$jiraServer/rest/api/*/issue/TEST-2?deleteSubTasks=True" -and $Method -eq "Delete"} {
+            return $null
+        }
+
+        # Generic catch-all. This will throw an exception if we forgot to mock something.
+        Mock Invoke-JiraMethod -ModuleName JiraPS {
+            ShowMockInfo 'Invoke-JiraMethod' 'Method', 'Uri'
+            throw "Unidentified call to Invoke-JiraMethod"
+        }
+
+        #############
+        # Tests
+        #############
+
+        Context "Sanity checking" {
+            $command = Get-Command -Name Remove-JiraIssue
+
+            defParam $command 'Issue'
+            defParam $command 'IncludeSubTasks'
+            defParam $command 'Credential'
+        }
+
+        Context "Functionality" {
+
+            It "Accepts generic object with the correct properties" {
+                {
+                    $issue = Get-JiraIssue -Key TEST-1
+                    Remove-JiraIssue -Issue $issue -Force
+                } | Should Not Throw
+                Assert-MockCalled -CommandName Invoke-JiraMethod -Exactly -Times 1 -Scope It
+            }
+
+            It "Accepts a JiraPS.Issue object over the pipeline" {
+                { Get-JiraIssue -Key TEST-1 | Remove-JiraIssue -Force} | Should Not Throw
+                Assert-MockCalled -CommandName Invoke-JiraMethod -Exactly -Times 1 -Scope It
+            }
+
+            It "Throws on issues with subtasks" {
```

Comment:
it shouldn't throw.
IMHO it should be a non-terminating error.

as in: 
```
Get-JiraIssue -Query "project= Foo" | Remove-JiraIssue -ErrorVariable $fails
```
I would expect the command to process all issues... (continuing if one fails)

Reply:
Right - normally, it would simply write a non-terminating error. However, it seems that Pester can't (directly) check if something wrote an error (at least according to pester/Pester#366) - the only options are to promote it to a terminating error using `-ErrorAction Stop` or to use `-ErrorVariable` and check its contents afterwards. I chose the former only because it seemed more concise.

Reply:
That's fine. but the text of the `It` block should describe what the function does. In this case, `writes an error on issues with subtasks`

Reply:
This was fixed in commit 2e86d9ffda63b7b04abd12cac0093ea24dc152c9.